src/java.desktop/macosx/native/libawt_lwawt/java2d/metal/MTLLayer.m

```diff
@@ -74,12 +80,38 @@ - (id) initWithJavaLayer:(jobject)layer
     self.framebufferOnly = NO;
     self.nextDrawableCount = 0;
     self.opaque = YES;
-    CVDisplayLinkCreateWithActiveCGDisplays(&displayLink);
-    CVDisplayLinkSetOutputCallback(displayLink, &displayLinkCallback, (__bridge void*)self);
-    self.displayLinkCount = 0;
+    [self createDisplayLink];
+
+    [[[NSWorkspace sharedWorkspace] notificationCenter]
+        addObserver : self
+           selector : @selector(onScreenSleep)
+               name : NSWorkspaceScreensDidSleepNotification object: NULL];
+
+    [[[NSWorkspace sharedWorkspace] notificationCenter]
+        addObserver : self
+           selector : @selector(onScreenWakeup)
+               name : NSWorkspaceScreensDidWakeNotification object: NULL];
+
     return self;
 }
 
+- (void) onScreenSleep {
+    J2dTraceLn(J2D_TRACE_VERBOSE, "MTLLayer.onScreenSleep ---  received screen sleep notification.");
+
+    [self stopDisplayLink];
+}
```

Comment:
I'm assuming this is something we should have done already to save wasted work ?
i.e if the screen is asleep, we don't need to be connected to it and updating windows ?
But I see the same stopDisplayLink call in the wakeup code ... any issues with the extra call there ?

Reply:
> I'm assuming this is something we should have done already to save wasted work ?
i.e if the screen is asleep, we don't need to be connected to it and updating windows ?

We start the DisplayLink from Metal RenderQueue whenever there is something to be painted from back buffer to the window buffer. Once the window buffer is painted in a DisplayLink callback, we stop the DisplayLink. 
We were not stopping the DisplayLink on screen sleep event. This is what this PR adds.

> But I see the same stopDisplayLink call in the wakeup code ... any issues with the extra call there ?

I have added that call as a sanity before releasing the Displaylink. It will be a no-op if DisplayLink is not running.

Reply:
ok